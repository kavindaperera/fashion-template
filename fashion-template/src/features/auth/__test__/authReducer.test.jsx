import * as types from '../authConstants';
import reducer from '../authReducer';
import { initialState  } from '../authReducer';

const mockUserData=[]
describe('auth reducer', () => {
    it('should return the initial state', () => {
      expect(reducer(undefined, {})).toEqual(
        {
            currentUser: {}
        }
      )
    })
    it('should handle SIGN_OUT_USER', () => {
        expect(
          reducer(initialState , {
            type: types.SIGN_OUT_USER,
            payload: {
                authenticated: false,
                currentUser: {}
              }
          })
        ).toEqual({
            authenticated: false,
            currentUser: {}
          })
      })

      it('should handle SIGN_OUT_USER', () => {
        expect(
          reducer(initialState , {
            type: types.SIGN_OUT_USER,
            payload: {
                authenticated: true,
                currentUser: {mockUserData}
              }
          })
        ).toEqual({
            authenticated: false,
            currentUser: {}
          })
      })

      //SnapShot testing

      it('should handle SIGN_OUT_USER', () => {
        expect(
          reducer(initialState , {
            type: types.SIGN_OUT_USER,
            payload: {
                authenticated: false,
                currentUser: {}
              }
          })
        ).toMatchSnapshot()
    })

    it('should handle SIGN_OUT_USER', () => {
        expect(
          reducer(initialState , {
            type: types.SIGN_OUT_USER,
            payload: {
                authenticated: true,
                currentUser: {mockUserData}
              }
          })
        ).toMatchSnapshot()
    })

    it('should handle SIGN_OUT_USER', () => {
        expect(
          reducer(initialState , {
            type: types.SIGN_OUT_USER,
            payload: {
                authenticated: false,
                currentUser: {}
              }
          })
        ).toMatchSnapshot()
    })
})