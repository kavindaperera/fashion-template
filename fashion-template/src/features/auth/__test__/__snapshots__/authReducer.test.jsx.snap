// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth reducer should handle SIGN_OUT_USER 1`] = `
Object {
  "authenticated": false,
  "currentUser": Object {},
}
`;

exports[`auth reducer should handle SIGN_OUT_USER 2`] = `
Object {
  "authenticated": false,
  "currentUser": Object {},
}
`;

exports[`auth reducer should handle SIGN_OUT_USER 3`] = `
Object {
  "authenticated": false,
  "currentUser": Object {},
}
`;
